### PR TITLE
Remove debug overhead on ACL

### LINKED ISSUE #558: Excessive ACL debug information
Since #550, we are getting excessive debug information:

```
  solid:ACL Can https://localhost:8443/profile/card#me Read https://localhost:8443/profile/card? +3m
  solid:ACL Check if ACL exists: https://localhost:8443/profile/card.acl +0ms
  solid:ACL Read access permitted to https://localhost:8443/profile/card#me +26ms
  solid:ACL Can an agent Read https://localhost:8443/profile/card? +3ms
  solid:ACL Can an agent Write https://localhost:8443/profile/card? +1ms
  solid:ACL Can an agent Append https://localhost:8443/profile/card? +0ms
  solid:ACL Can an agent Control https://localhost:8443/profile/card? +0ms
  solid:ACL Can https://localhost:8443/profile/card#me Read https://localhost:8443/profile/card? +0ms
  solid:ACL Can https://localhost:8443/profile/card#me Write https://localhost:8443/profile/card? +0ms
  solid:ACL Can https://localhost:8443/profile/card#me Append https://localhost:8443/profile/card? +0ms
  solid:ACL Can https://localhost:8443/profile/card#me Control https://localhost:8443/profile/card? +0ms
  solid:ACL Read access permitted to null +0ms
  solid:ACL Write access NOT permitted to null +1ms
  solid:ACL Append access NOT permitted to null +1ms
  solid:ACL Control access NOT permitted to null +0ms
  solid:ACL Read access permitted to https://localhost:8443/profile/card#me +0ms
  solid:ACL Write access permitted to https://localhost:8443/profile/card#me +0ms
  solid:ACL Append access permitted to https://localhost:8443/profile/card#me +0ms
  solid:ACL Control access permitted to https://localhost:8443/profile/card#me +0ms
  solid:ACL Write access denied to null +1ms
  solid:ACL Error: ACL file found but no matching policy found
  solid:ACL     at permissionSet.checkAccess.then.hasAccess (/Users/ruben/Documents/UGent/MIT/Solid/node-solid-server/lib/acl-checker.js:101:17)
  solid:ACL     at <anonymous> +0ms
  solid:ACL Append access denied to null +0ms
  solid:ACL Error: ACL file found but no matching policy found
  solid:ACL     at permissionSet.checkAccess.then.hasAccess (/Users/ruben/Documents/UGent/MIT/Solid/node-solid-server/lib/acl-checker.js:101:17)
  solid:ACL     at <anonymous> +0ms
  solid:ACL Control access denied to null +1ms
  solid:ACL Error: ACL file found but no matching policy found
  solid:ACL     at permissionSet.checkAccess.then.hasAccess (/Users/ruben/Documents/UGent/MIT/Solid/node-solid-server/lib/acl-checker.js:101:17)
  solid:ACL     at <anonymous> +0ms
  solid:ACL Error: ACL file found but no matching policy found +1ms
  solid:ACL Authentication required +0ms
  solid:ACL Error: ACL file found but no matching policy found +0ms
  solid:ACL Authentication required +0ms
  solid:ACL Error: ACL file found but no matching policy found +0ms
  solid:ACL Authentication required +0ms
```

This is because all 4 modes are checked for the current user and the public.

We should move the logging such that only the requested permissions are logged, and the others simply summarized.

### PR DESCRIPTION
**Pull request by RubenVerborgh**
Fixes #558, the debug output for the same action is now:

```
  solid:ACL Check if ACL exists: https://localhost:8443/profile/card.acl +6s
  solid:ACL Permissions for (none): read +28ms
  solid:ACL Permissions for public: read +0ms
  solid:get /profile/card on localhost +1ms
  solid:get HEAD only +1ms
```

### CODE REVIEW

File: lib/acl-checker.js [hunk 1]
```diff
@@ -36,13 +35,10 @@ class ACLChecker {
...
     return this._permissionSet
       .then(acls => this.checkAccess(acls, user, mode))
-      .catch(err => {
-        debug(`Error: ${err.message}`)
+      .catch(() => {
         if (!user) {
-          debug('Authentication required')
           throw new HTTPError(401, `Access to ${this.resource} requires authorization`)
         } else {
-          debug(`${mode} access denied for ${user}`)
```
Comment (dan-f) [inline lib/acl-checker.js]: This seems like a useful message, no?

File: lib/handlers/allow.js [hunk 2]
```diff
@@ -36,8 +37,12 @@ function allow (mode) {
       })
 
       // Ensure the user has the required permission
-      req.acl.can(req.session.userId, mode)
-             .then(() => next(), next)
+      const userId = req.session.userId
+      req.acl.can(userId, mode)
+        .then(() => next(), err => {
```
Comment (dan-f) [inline lib/handlers/allow.js]: Nitpick: can replace `() => next()` with `next`

Comment (RubenVerborgh) [reply lib/handlers/allow.js]: We can't: the promise resolves to a non-falsy value (`true`), and the first parameter of `next` being truthy is interpreted as an error.

File: lib/acl-checker.js (hunk 1 above)
Comment (RubenVerborgh) [reply lib/acl-checker.js]: It is, just not 8 times :wink: That's why I moved it here: https://github.com/solid/node-solid-server/pull/566/files#diff-b11433f1fb19c85e170915d27502d490R43

File: lib/handlers/allow.js (hunk 2 above)
Comment (dan-f) [reply lib/handlers/allow.js]: 🤦‍♂️ 🤦‍♀️

**Pull request merged**